data/src/main/java/org/akvo/flow/data/repository/FormDataRepository.kt

```diff
@@ -94,6 +105,63 @@ class FormDataRepository @Inject constructor(
         )
     }
 
+    override suspend fun processZipFile(file: File, instanceUrl: String, awsBucket: String) {
+        val zipFile = ZipFile(file)
+        val entries: Enumeration<out ZipEntry> = zipFile.entries()
+        while (entries.hasMoreElements()) {
+            val entry = entries.nextElement()
+            val entryName = entry.name ?: ""
+            when {
+                entryName.endsWith(BootstrapProcessor.CASCADE_RES_SUFFIX) -> {
+                    processCascadeResource(zipFile, entry)
+                }
+                entryName.endsWith(BootstrapProcessor.XML_SUFFIX) -> {
+                    processSurveyFile(zipFile, entry, instanceUrl, awsBucket)
+                }
+            }
+        }
+        zipFile.close()
+        file.renameTo(File(file.absolutePath + BootstrapProcessor.PROCESSED_OK_SUFFIX))
+    }
+
+    private fun processSurveyFile(
+        zipFile: ZipFile,
+        entry: ZipEntry,
+        instanceUrl: String,
+        awsBucket: String,
+    ) {
+        val inputStream = zipFile.getInputStream(entry)
+        val formAndMeta = dataFormMapper.mapFormAndMetadata(xmlParser.parseXmlFormWithMeta(inputStream))
+
+        //verify form
+        val surveyMetadata = formAndMeta.second
+        if (surveyMetadata.alias.isNotEmpty()) {
+            if (!instanceUrl.contains(surveyMetadata.alias)) {
+                throw WrongDashboardError()
+            }
+        } else if (surveyMetadata.app.isNotEmpty()) {
+            if (!awsBucket.contains(surveyMetadata.app)) {
+                throw WrongDashboardError()
+            }
+        } else {
+            throw WrongDashboardError()
+        }
```

Comment:
I was wondering about this section. Shouldnt they be multiple `if` statements rather than `if.. else`.  Say for instance alias is not empty, and is correct but the app info is incorrect, that will be skipped.  I guess the assumption is that once one is correct then the other should be but for example the recent migrations of instances may invalidate that assumption.

Reply:
alias has priority over app, that is why we added it so in this case app does not matter